### PR TITLE
fix: snapshots have to be plain objects

### PR DESCRIPTION
**Pull request by dai-shi**
continuing from #24 

As of now, proxy-compare does only support plain objects and arrays, we can't use prototypes.
This is to work around by copying keys.

I'm not confident that test cases cover everything. Help wanted.

### CODE REVIEW

**Comment by dai-shi** [comment]
`Class TicTac` still seems to be working. https://codesandbox.io/s/valtio-tic-tac-forked-p14i1?file=/src/index.js

(Note: This example doesn't do well for re-renders. Modified version: https://codesandbox.io/s/valtio-tic-tac-forked-nwp42?file=/src/index.js )

File: src/vanilla.ts [hunk 1]
```diff
@@ -50,10 +53,21 @@ export const proxy = <T extends object>(initialObject: T = {} as T): T => {
...
-          ? []
-          : Object.create(target.constructor?.prototype || null)
+        const snapshot: any = Array.isArray(target) ? [] : {}
         snapshotCache.set(receiver, { version, snapshot })
+        Reflect.ownKeys(Object.getPrototypeOf(target) || {}).forEach((key) => {
+          if (key === Symbol.unscopables) return
+          try {
+            const value = target[key]
+            // if (typeof value === 'function') {
+            //   snapshot[key] = value.bind(target)
```
Comment (dai-shi) [inline src/vanilla.ts:63]: I couldn't make a test case that would require this bind.

**Comment by dai-shi** [comment]
I still don't think it's optimal and there must be various edge cases, but let's merge and release this. Feedbacks are welcome.
